google/cloud/spanner/ci/lib/spanner_emulator.sh

```diff
@@ -83,6 +102,36 @@ function spanner_emulator::start() {
 
   export SPANNER_EMULATOR_HOST="localhost:${emulator_port}"
   io::log "Spanner emulator started at ${SPANNER_EMULATOR_HOST}"
+
+  # Repeat the process to launch the emulator with HTTP support. We launch a separate process
+  # as existing tests fail if we try and use gateway_main for both gRPC and REST.
+  readonly SPANNER_HTTP_EMULATOR_CMD="${CLOUD_SDK_LOCATION}/bin/cloud_spanner_emulator/gateway_main"
+  echo "${SPANNER_HTTP_EMULATOR_CMD}"
+  if [[ ! -x "${SPANNER_HTTP_EMULATOR_CMD}" ]]; then
+    echo 1>&2 "The Cloud Spanner HTTP emulator does not seem to be installed, aborting"
+    return 1
+  fi
+
+  # The tests typically run in a Docker container, where the ports are largely
+  # free; when using in manual tests, you can set REST_EMULATOR_PORT.
+  rm -f http_emulator.log
+  # Leverage the previously determined emulator_port to avoid collision.
+  local http_port=$((emulator_port + 1))
+  local grpc_port=$((http_port + 1))
+  "${SPANNER_HTTP_EMULATOR_CMD}" --hostname "localhost" --grpc_port "${grpc_port}" --http_port "${http_port}" >http_emulator.log 2>&1 </dev/null &
+  SPANNER_HTTP_EMULATOR_PID=$!
+
+  http_emulator_port="$(spanner_emulator::internal::read_http_emulator_port http_emulator.log)"
```

Comment:
For `emulator_main` we re-used the local `emulator_port`.  Should we just do the same with `http_port` here?  Or rename `http_port` to `http_emulator_port` (and `grpc_port`)?